assets/js/components/ChargingPlanRepeatingSettings.vue

```diff
@@ -0,0 +1,233 @@
+						:checked="selectedActive"
+						@change="update(true)"
+					/>
+				</div>
+			</div>
+			<div class="col-1 mx-auto d-flex align-items-center justify-content-start">
+				<button
+					v-if="showApply"
+					type="button"
+					class="btn btn-sm btn-outline-primary border-0 text-decoration-underline"
+					data-testid="repeating-plan-apply"
+					@click="update(true)"
+				>
+					{{ $t("main.chargingPlan.update") }}
+				</button>
+				<button
+					v-else
+					type="button"
+					class="btn btn-sm btn-outline-secondary border-0"
+					data-testid="repeating-plan-delete"
+					@click="$emit('repeating-plan-removed', id)"
+				>
+					<shopicon-regular-trash size="s" class="flex-shrink-0"></shopicon-regular-trash>
+				</button>
+			</div>
+		</div>
+	</div>
+</template>
+
+<script>
+import "@h2d2/shopicons/es/regular/trash";
+import { distanceUnit } from "../units";
+import MultiSelect from "./MultiSelect.vue";
+import formatter from "../mixins/formatter";
+
+export default {
+	name: "ChargingPlanRepeatingSettings",
+	components: {
+		MultiSelect,
+	},
+	mixins: [formatter],
+	props: {
+		id: Number,
+		weekdays: { type: Array, default: () => [] },
+		time: String,
+		soc: Number,
+		active: Boolean,
+		rangePerSoc: Number,
+		formIdPrefix: String,
+		numberPlans: Boolean,
+		dataChanged: Boolean,
+	},
+	emits: ["repeating-plan-updated", "repeating-plan-removed"],
+	data: function () {
+		return {
+			selectedWeekdays: this.weekdays,
+			selectedTime: this.time,
+			selectedSoc: this.soc,
+			selectedActive: this.active,
+		};
+	},
+	computed: {
+		showApply: function () {
+			return this.dataChanged && this.selectedActive;
+		},
+		weekdaysLabel: function () {
+			return this.getShortenedWeekdaysLabel(this.selectedWeekdays);
+		},
+		socOptions: function () {
+			// a list of entries from 5 to 100 with a step of 5
+			return Array.from(Array(20).keys())
+				.map((i) => 5 + i * 5)
+				.map(this.socOption);
+		},
+		dayOptions: function () {
+			return this.getWeekdaysList("long");
+		},
+		number: function () {
+			return this.id + 2;
+		},
+	},
+	watch: {
+		weekdays(newValue) {
+			this.selectedWeekdays = newValue;
+		},
+		time(newValue) {
+			this.selectedTime = newValue;
+		},
+		soc(newValue) {
+			this.selectedSoc = newValue;
+		},
+		active(newValue) {
+			this.selectedActive = newValue;
+		},
+	},
+	methods: {
+		changeSelectedWeekdays: function (weekdays) {
+			this.selectedWeekdays = weekdays;
+			this.update();
+		},
+		formId: function (name) {
+			return `${this.formIdPrefix}-${this.id}-${name}`;
+		},
+		socOption: function (value) {
+			const name = this.fmtSocOption(value, this.rangePerSoc, distanceUnit());
+			return { value, name };
+		},
+		update: function (save) {
+			this.$emit("repeating-plan-updated", {
+				id: this.id,
+				save: !this.selectedActive || true === save,
+				weekdays: this.selectedWeekdays,
+				time: this.selectedTime,
+				soc: this.selectedSoc,
+				active: this.selectedActive,
+			});
```

Comment:
Vorschlag:

```suggestion
			this.$emit("repeating-plan-updated", {
				id: this.id,
				save: !this.selectedActive || true === save,
				plan: {
					weekdays: this.selectedWeekdays,
					time: this.selectedTime,
					soc: this.selectedSoc,
					active: this.selectedActive,
				}
			});
```

Wenn du die Plan-Werte hier zusammenhälst macht es das bei der Event-Verarbeitung deutlich einfacher. Da gibts mehrere Stellen, wo du wieder neue Objekte aus (weekdays, time, soc, active) baust, um die Metadaten `id` und `save` nicht per API zu verschicken.